### PR TITLE
Expose HTTP API for parsing files and listing supported languages

### PR DESCRIPTION
**Pull request by dennwc**
Currently, the only way to talk to bblfshd is via gRPC API. This means that user needs to install one of our CLIs from one of the language clients in order to test it.

This PR adds a simple HTTP API for parsing files, listing supported languages and checking the version.

**Version:**
```
GET /api/v1/version
```
or
```sh
curl http://localhost:9480/api/v1/version
```

**Supported languages:**
```
GET /api/v1/languages
```
or
```sh
curl http://localhost:9480/api/v1/languages
```

**Parse a file:**
```
POST /api/v1/parse?lang=xxxx
Accept: application/json

<file content>
```
or
```sh
curl --data-binary @main.go -H 'Accept: application/json' 'http://localhost:9480/api/v1/parse?lang=go'
```

Supported formats for the parse response:
- `application/json`
- `application/x-yaml`
- `application/octet-stream` (our binary encoding)

Signed-off-by: Denys Smirnov <denys@sourced.tech>

### CODE REVIEW

**Comment by juanjux** [comment]
I don't think installing a CLI or using `bblfshctl` is a big deal, and it was decided that gRPC was the protocol we would use, plus this adds another protocol to maintain & debug & document, but the implementation is really simple so LGTM but until we've a lead we should probably leave the decision to add a new protocol to bblfshd to @smola.

**Comment by dennwc** [comment]
To the record, this was done to remove the backend used by the Dashboard.

**Comment by smola** [comment]
This would be pretty useful and code is pretty simple (it would require tests, though).
Added @mcuadros.

**Comment by mcuadros** [comment]
Why we are having a JSON interface?

**Comment by dennwc** [comment]
@mcuadros To serve it for Babelfish Dashboard. Also it's nice for user because you can just `curl` the file to bblfshd and save result as JSON/YAML.

File: cmd/bblfshd/main.go [hunk 1]
```diff
@@ -150,6 +178,195 @@ func listenControl(d *daemon.Daemon) {
...
+		}
+	}
+
+	maxSize := *maxMessageSize * 1024 * 1024
+	data, err := ioutil.ReadAll(io.LimitReader(r.Body, int64(maxSize+1)))
+	if err != nil {
+		jsonError(w, http.StatusBadRequest, err)
+		return
+	} else if len(data) > maxSize {
+		jsonError(w, http.StatusBadRequest, fmt.Errorf("content is too large"))
```
[the remark this answers is not in the document]
Comment (dennwc) [reply cmd/bblfshd/main.go]: Good point!

**Comment by dennwc** [comment]
@mcuadros Friendly ping

**Comment by bzz** [comment]
I belive this action item from TL meeting might be relevant https://github.com/src-d/minutes/pull/342/files#diff-e451e6c5126356b0e298741d53c2ca44R31

**Comment by dennwc** [comment]
I want to hear specific details on why this is a "no go". @smola @mcuadros @bzz Can you elaborate?

**Comment by bzz** [comment]
Sorry @dennwc - I have no idea. Just stumbled up on this in minutes and figured out it might be relevant to this PR so posted the link.

**Comment by mcuadros** [comment]
@dennwc is a topic that we personally already discussed and even with that was developed, without any design documents or discussions

Having a HTTP interface introduce complexity in the communication of the project, maintainership and many other places, the chosen interface for bblfsh is gRPC aswell as his representations. So having JSON representation and HTTP endpoint should be avoided. 

Nowadays we have HTTP proxy for bblfsh-web, just because JavaScript sucks, and didn't had proper support for gRPC, as soon as a proper gRPC implementation can be used in bblfsh-web, the proxy should be deleted and burn.

**Pull request closed without merging**